### PR TITLE
Cleanup bucket bar

### PR DESCRIPTION
**Pull request by esanzgar**
- Added types
- Moved out of the plugin directory
- Avoid extending from Delegator (unnecessary)

### CODE REVIEW

File: src/annotator/bucket-bar.js [hunk 1]
```diff
@@ -1,24 +1,30 @@
...
 import { render } from 'preact';
-import Buckets from '../components/Buckets';
+import Buckets from './components/Buckets';
+
+import { anchorBuckets } from './util/buckets';
 
-import { anchorBuckets } from '../util/buckets';
+/**
+ * @typedef BucketBarOptions
+ * @prop {string[]} scrollables  // Selectors for the scrollable elements on the page
```
Comment (robertknight) [inline src/annotator/bucket-bar.js:8]:
```suggestion
 * @prop {string[]} scrollables - Selectors for the scrollable elements on the page
```

File: src/annotator/bucket-bar.js [hunk 2]
```diff
@@ -1,24 +1,30 @@
...
+   * @param {Pick<import('./guest').default, 'anchors'|'selectAnnotations'>} guest
+   */
+  constructor(container, options, guest) {
     const defaultOptions = {
-      // Selectors for the scrollable elements on the page
       scrollables: [],
     };
 
-    const opts = { ...defaultOptions, ...options };
+    this.options = { ...defaultOptions, ...options };
```
Comment (robertknight) [inline src/annotator/bucket-bar.js:22]: There is no real need to have `this.options` any more. You could just have a `this.scrollables` argument which is initialized to an empty array if there is no `scrollables` property in `options`.

**Review by robertknight — COMMENTED** [review]
I think this is is going in the right direction. There are some remaining references to `BucketBar` being a "plugin" in other parts of the code, eg. in `guest.js`, `sidebar.js` and tests. Do you have a plan for those?

File: src/annotator/bucket-bar.js (hunk 2 above)
Comment (esanzgar) [reply src/annotator/bucket-bar.js:22]: I addressed this as you suggested.

**Comment by esanzgar** [comment]
> I think this is is going in the right direction. There are some remaining references to BucketBar being a "plugin" in other parts of the code, eg. in guest.js, sidebar.js and tests. Do you have a plan for those?

I addressed that in a second commit in this PR.

File: src/annotator/guest.js [hunk 3]
```diff
@@ -151,6 +151,7 @@ export default class Guest extends Delegator {
     });
 
     this.plugins = {};
+    this.BucketBar = null;
```
Comment (robertknight) [inline src/annotator/guest.js:154]:
```suggestion
    /** @typedef {BucketBar|null} */
    this.bucketBar = null;
```

No reason not to use our normal public field naming conventions here. Also its helpful for readers to specify the type.

File: src/annotator/test/bucket-bar-test.js [hunk 4]
```diff
@@ -10,7 +10,7 @@ describe('BucketBar', () => {
 
   const createBucketBar = function (options) {
     const element = document.createElement('div');
-    return new BucketBar(element, options || {}, fakeAnnotator);
+    return new BucketBar(element, fakeAnnotator, options);
```
Comment (robertknight) [inline src/annotator/test/bucket-bar-test.js:13]: 👍 - swapping the options object to be the last parameter is nicer.

**Review by robertknight — APPROVED** [review]
Hi Eduardo - The changes here look good. I suggested to change the name of the `BucketBar` field to `bucketBar` and document its type to fit with our normal conventions for public fields. Please make sure you give this a test with both PDF and HTML documents.

File: src/annotator/guest.js (hunk 3 above)
Comment (esanzgar) [reply src/annotator/guest.js]: Awesome.

**Pull request merged**